tests/pack.rs

```diff
@@ -2,35 +2,84 @@ use anyhow::Result;
 
 mod common;
 
-use common::fixture::*;
+use common::blk_archive::PackResponse;
+use common::fixture::{create_archive, create_input_file, BLOCK_SIZE};
+use common::random::Pattern;
 use common::test_dir::*;
 
 //-----------------------------------------
 
 #[test]
 fn pack_one_file() -> Result<()> {
     let mut td = TestDir::new()?;
-    let archive = create_archive(&mut td)?;
+    let archive = create_archive(&mut td, true)?;
 
     let file_size = 16 * 1024 * 1024;
     let seed = 1;
-    let input = create_input_file(&mut td, file_size, seed)?;
-    let stream = archive.pack(&input)?;
+    let input = create_input_file(&mut td, file_size, seed, Pattern::LCG)?;
+    let stream = archive.pack(&input)?.stream_id;
     archive.verify(&input, &stream)
 }
 
 #[test]
 fn pack_same_file_multiple_times() -> Result<()> {
     let mut td = TestDir::new()?;
-    let archive = create_archive(&mut td)?;
+    let archive = create_archive(&mut td, true)?;
 
     let file_size = 16 * 1024 * 1024;
     let seed = 1;
-    let input = create_input_file(&mut td, file_size, seed)?;
+    let input = create_input_file(&mut td, file_size, seed, Pattern::LCG)?;
     let streams = (0..3)
         .map(|_| archive.pack(&input))
         .collect::<Result<Vec<_>>>()?;
-    streams.iter().try_for_each(|s| archive.verify(&input, s))
+    streams
+        .iter()
+        .try_for_each(|s| archive.verify(&input, &s.stream_id))
+}
+
+fn pack_common_verify_stats(file_size: u64, pattern: Pattern) -> Result<PackResponse> {
+    let mut td = TestDir::new()?;
+    let seed = 1;
+
+    let archive = create_archive(&mut td, false)?;
+    let input = create_input_file(&mut td, file_size, seed, pattern)?;
+
+    let data_start = archive.data_size()?;
+    let response = archive.pack(&input)?;
+    let data_end = archive.data_size()?;
+
+    let data_written = data_end - data_start;
+
+    // Amount on disk will be > what we report to user as we don't include slab metadata
+    let diff = data_written - response.stats.data_written;
+    let approx_expected_meta = (response.stats.data_written / 4 * 1024 * 1024) * 24;
+    assert!(diff <= approx_expected_meta);
```

Comment:
I wonder if this assertion could be expressed more directly, to make the tested behavior easier to understand. As the archive is not compressed, maybe we could simply verify the data_written returned matches the stream length?

Reply:
> simply verify the data_written returned matches the stream length?

I'm confused by this statement as the stream length is the number of bytes that comprise the encoded stream which is vastly different than the number of bytes written to the archive?

Reply:
oh, here the stream length I mean is the input file size, not the encoded stream.

Reply:
This logic is used in a common function that verifies numbers which would fail in the case we are testing zero bytes or same byte unit tests.

Reply:
@mingnus  No more calculations, straight number comparison as suggested, thanks!

Reply:
Thanks. I think this is a better approach, since the caller shouldn't assume how much data or metadata is written without knowing the data pattern.